src/sequences/OEIS.ts

```diff
@@ -40,131 +18,160 @@ const paramDesc = {
+        return this.oeisName
     }
 
     /* Unlike the base Cached sequence class, we grab the entire sequence
-       at once.
+       at once. Not only that, but we maintain a global cache for all OEIS
+       sequences.
     */
-    async fillValueCache(): Promise<void> {
-        // Catch HTTP errors. (This function has multiple HTTP requests.)
-        try {
-            // import.meta.env is basically your configuration.
-            // It's set up by Vite automatically
-            // See https://vitejs.dev/guide/env-and-mode.html.
-            const urlPrefix = `${import.meta.env.VITE_BACKSCOPE_URL}/api/`
-            const backendUrl =
-                urlPrefix
-                + `get_oeis_values/${this.oeisId}/${this.cacheBlock}`
-            const response = await axios.get(backendUrl)
-            this.first = Infinity
-            this.last = -Infinity
-            for (const k in response.data.values) {
-                const index = Number(k)
-                if (index < this.first) this.first = index
-                if (index > this.last) this.last = index
-                this.cache[index] = BigInt(response.data.values[k])
-                if (this.modulus) {
+    static vCache: Record<string, OEISdata> = {}
+    async fillValueCache(_n?: number) {
+        // If the cache is already full, nothing to do:
+        if (this.lastValueCached >= this.last) return
+        let known = this.oeisID in OEIS.vCache
+        if (!known) {
+            // Catch HTTP errors. (This function has multiple HTTP requests.)
+            try {
+                const backendUrl =
+                    OEIS._urlPrefix
+                    + `get_oeis_name_and_values/${this.oeisID}`
+                const response = await axios.get(backendUrl)
+                let name = response.data.name
+                if (this.oeisID == 'A000045') name = 'Virahāṅka-' + name
+                let first = Infinity
+                let last = -Infinity
+                const values: bigint[] = []
+                for (const k in response.data.values) {
+                    const index = Number(k)
+                    if (index < first) first = index
+                    if (index > last) last = index
+                    values[index] = BigInt(response.data.values[k])
+                }
+                OEIS.vCache[this.oeisID] = {name, first, last, values}
+            } catch (e) {
+                window.alert(alertMessage(e))
+            }
+        }
+        known = this.oeisID in OEIS.vCache
+        if (!known) {
+            this.oeisName = `Unknown OEIS sequence ${this.oeisID}`
+            this.first = 0
+            this.last = -1
+            this.cacheBlock = 0
+            // Anything else we should do? Throw an error?
+        } else {
+            const data = OEIS.vCache[this.oeisID]
+            this.oeisName = data.name
+            this.first = data.first
+            this.last = data.last
```

Comment:
> The OEIS has 100,000 primes but we are only loading 10,000 max. In Chaos, for example, it makes a much better picture with 100,000.

That's because the OEIS "b-file" only has 10000 primes. There is a separate, idiosyncratic file on the OEIS with 100000 primes. We could make a special rule in the backend to grab that one. Or, if we are making a special rule anyway, we could just use ompoasso.free.fr/primelistweb that has over 50 million primes... in any case, I will file this as a backend issue right now.

Reply:
> Actually, we seem to be getting fewer than 10000 primes. When I do ModFill with maximum modulus 10000, it doesn't fill up the triangle.

That's because ModFill has an arbitrary internally hard-coded index limit of 1000, which would be easy to remove. Anyhow, if you try some other visualizer, say NumberGlyph if you are willing to wait a long time, you can see that we are getting all 10000 terms.

Reply:
> If we want to have special "built-in" primes and Vir-Fibs, do we want to actually program in functionality to create many more of these values than OEIS offers? This might go into the issues or post-delft todos, but it seemed relevant here so I'll mention it.

I don't know of any way to have more primes other than to just use a bigger list, which I have filed as a backscope issue. I agree that we should add more number-theoretic functions to mathjs, which will be easy to do; we should add virfib(n), which with bigints will be essentially unlimited (although may get slow for indices in the thousands), and virfibmod(n,m), which as you say will be unlimited. Whether either of these things mean we should change the behavior of OEIS A000045 is unclear to me, and I personally think that's a discussion for another time, not this PR.

Reply:
[We could even add a rule to mathjs that would collapse an expression like `virfib(n) % m` to `virfibmod(n,m)` -- probably a good idea to do that.]

Reply:
Ok, yes, this is a good backscope issue, it seems crazy to offer only 10000 primes :)  Resolving this conversation tho.